### ISSUE TITLE
[@typescript-eslint/no-unused-vars] false negative in for loop

### ISSUE THREAD

**Issue by Maxim-Mazurok** [typescript-eslint/typescript-eslint#888]
**Repro**

```JSON
{
  "extends": ["plugin:@typescript-eslint/recommended"]
}
```

```TS
for (const property in headers) {
  if (headers.hasOwnProperty(property)) {
    request.setRequestHeader(property, headers[property]);
  }
}
```

**Expected Result**

No warnings

**Actual Result**

`warning  'property' is defined but never used  @typescript-eslint/no-unused-vars`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.0.0` |
| `@typescript-eslint/parser`        | `2.0.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.2.1` |
| `node`                             | `10.16.3` |
| `npm`                              | `6.10.3` |

**Comment by bradzacher** [comment]
I am unable to reproduce this issue.
We extend the base eslint rule, and it looks like this case working fine:

https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVudXNlZC12YXJzOiAnd2FybicgKi9cbmZvciAoY29uc3QgcHJvcGVydHkgaW4gaGVhZGVycykge1xuICBpZiAoaGVhZGVycy5oYXNPd25Qcm9wZXJ0eShwcm9wZXJ0eSkpIHtcbiAgICByZXF1ZXN0LnNldFJlcXVlc3RIZWFkZXIocHJvcGVydHksIGhlYWRlcnNbcHJvcGVydHldKTtcbiAgfVxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTAsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7fSwiZW52Ijp7ImNvbW1vbmpzIjpmYWxzZSwibm9kZSI6ZmFsc2V9fX0=

**Comment by Maxim-Mazurok** [comment]
Sorry, it seems that the issue reproduces only with `parser: 'babel-eslint'` (10.0.2)

**Comment by bradzacher** [comment]
This sounds like a bug for that package.
Our rule just builds on top of the existing rule, so if you're experiencing a bug with `babel-eslint`, then it's a buggy interaction between the base rule and that parser.

**Comment by Maxim-Mazurok** [comment]
Ok, so should I open an issue in `babel-eslint` or in `eslint` repo?

**Comment by bradzacher** [comment]
I'm not sure exactly, because I don't know where the error is.
I would suggest starting with the `eslint` repo.

**Comment by Maxim-Mazurok** [comment]
Update: it seems to be resolved here: https://github.com/eslint/eslint/issues/12117#issuecomment-525128667
as of Eslint 6.2.2
